components/device-mgt-extensions/org.wso2.carbon.device.mgt.extensions.device.type.deployer/src/test/java/org/wso2/carbon/device/mgt/extensions/device/type/deployer/DeviceTypePluginDeployerTest.java

```diff
@@ -0,0 +1,80 @@
+package org.wso2.carbon.device.mgt.extensions.device.type.deployer;
+
+import org.apache.axis2.deployment.DeploymentException;
+import org.apache.axis2.deployment.repository.util.DeploymentFileData;
+import org.junit.Assert;
+import org.mockito.Mockito;
+import org.osgi.framework.ServiceRegistration;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.device.mgt.extensions.device.type.template.DeviceTypeConfigIdentifier;
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class DeviceTypePluginDeployerTest {
+
+    private DeviceTypePluginDeployer deviceTypePluginDeployer;
+    private DeploymentFileData deploymentFileData;
+
+    private Field deviceTypeServiceRegistrations = null;
+    private Field deviceTypeConfigurationDataMap = null;
+    private ServiceRegistration serviceRegistration = null;
+
+    @BeforeClass
+    public void init() throws Exception {
```

Comment:
Why are throwing generic exceptions? Is there any part which throw generic exception? If not shall we throw specific exceptions.